src/core/expr/fexpr_mean.cc

```diff
@@ -80,42 +80,42 @@ class FExpr_Mean : public FExpr_Func {
                           ));
         case SType::BOOL:
         case SType::INT8:
+          return make<int8_t, double>(std::move(col), gby, is_grouped);
         case SType::INT16:
-        case SType::INT32:        
+          return make<int16_t, double>(std::move(col), gby, is_grouped);
+        case SType::INT32:   
+          return make<int32_t, double>(std::move(col), gby, is_grouped);     
         case SType::INT64:
+          return make<int64_t, double>(std::move(col), gby, is_grouped);
         case SType::FLOAT64:
-          return make<double>(std::move(col), SType::FLOAT64, gby, is_grouped);
+          return make<double, double>(std::move(col), gby, is_grouped);
         case SType::FLOAT32:
-          return make<float>(std::move(col), SType::FLOAT32, gby, is_grouped);
-
+          return make<float, float>(std::move(col), gby, is_grouped);
         case SType::DATE32: {
-          Column coli = make<double>(std::move(col), SType::FLOAT64, gby, is_grouped);
+          Column coli = make<double, double>(std::move(col), gby, is_grouped);
```

Comment:
So why it is still not passing stype here and why do we need to pass additional type to `make`? I propose you revert the changes related to this switch, as we already discussed it will not work this way.

Reply:
take for instance `<int8_t, double>` - `int8_t` is the dtype of the input column, while `double` is the dtype of the output column. this is what makes up the template form `<T_IN, T_OUT>`, and in the reduce_unary.h file, in line 42, there is a `col_.can_be_read_as<T_IN>` and since int8_t is the dtype of the input column, there should be no issue. This allows us to pass dtype for scenarios where deriving the dtype from the input column might not be desirable. Say for example we are doing a count and we have a `FLOAT64` column, we shouldnt cast it to int64 to lose precision. instead we can do `<double, int64_t>` and reduce_unary.h will take care of the rest. As usual, I am open to feedback - helps with my understanding of the codebase. thanks @oleksiyskononenko

Reply:
This piece of code is not related to `count()`, so here we don't need to change anything.